src/main/java/io/openliberty/tools/common/plugins/config/LooseConfigData.java

```diff
@@ -25,13 +25,18 @@
 import org.w3c.dom.Element;
 
 public class LooseConfigData extends XmlDocument {
-    
+
     private String projectRoot = null;
+    private String sourceOnDiskName = "/devmode";
```

Comment:
I don't think the default value should be "/devmode" here, since this class is not specific to dev mode.  Supposedly we expect that `projectRoot` and `sourceOnDiskName` are either both set, or both not set.  Either somehow enforce that rule or change line 113 to also check if `sourceOnDiskName` is set, otherwise fallback to the canonical path using line 116.

Reply:
I think it is clearest if we initialize both fields to null and check on line 113 to make sure both have been set to definitive values. I'll also comment both setters to make this clear.